components/automate-ui/src/app/services/http/http-client-auth.interceptor.ts

```diff
@@ -49,13 +47,23 @@ export class HttpClientAuthInterceptor implements HttpInterceptor {
     if (this.projects && filtered) {
       headers = headers.set('projects', this.projects);
     }
-    return next
-      .handle(request.clone({ headers, params })).pipe(
-        catchError((error: HttpErrorResponse) => {
-          if (error.status === 401) {
-            this.chefSession.logout();
-          }
-          return observableThrowError(error);
-        }));
+
+    return this.chefSession.token_provider.pipe(
+      take(1),
+      mergeMap(id_token => {
+        // TODO(sr): Can we check if the request is sent to some external API?
+        //           Right now, we're throwing the ID token across the internet for telemetry,
+        //           where it's not needed. It would be nice to _not_ do that.
+        headers = headers.set('Authorization', `Bearer ${id_token}`);
+        return next
+          .handle(request.clone({ headers, params })).pipe(
+            catchError((error: HttpErrorResponse) => {
+              if (error.status === 401) {
+                this.chefSession.logout();
+              }
+              return observableThrowError(error);
```

Comment:
Is this the cleanest thing we can do? IIRC, `this.chefSession.logout()` will change the location (go somepace else) directly, so may I ask for a comment indicating that? For the casual reader, there's no way to tell that _this line_ will not be reached when the status is 401.

Reply:
👍 We probably should return as well? I'm not sure if changing the location stops the execution of the javascript function?

Reply:
It'll go to dex, so the JS code here is unloaded. I don't know how that looks in detail, though.